src/utils/AnalyticsUtils.ts

```diff
@@ -1,13 +1,64 @@
+import { history } from 'coveo.analytics';
 import { IAnalyticsActionCause, analyticsActionCauseList } from '../ui/Analytics/AnalyticsActionListMeta';
+import { findLastIndex } from 'underscore';
+
+export interface IClientInformationProvider {
+  readonly visitorId: string;
+  readonly clientId: string;
+  readonly disabled: boolean;
+}
 
 export class AnalyticsUtils {
-  static addActionCauseToList(newActionCause: IAnalyticsActionCause) {
+  private static currentInstance: AnalyticsUtils;
+  private clientInformationProvider: IClientInformationProvider = null;
+
+  public static get instance() {
+    return this.currentInstance || (this.currentInstance = new AnalyticsUtils());
+  }
+
+  private constructor() {}
+
+  public get visitorId() {
+    return this.trackingEnabled ? this.clientInformationProvider.visitorId : null;
+  }
+
+  public get clientId() {
+    return this.trackingEnabled ? this.clientInformationProvider.clientId : null;
+  }
+
+  public get pageId() {
+    const store = new history.HistoryStore();
+    const actions = store.getHistory() as { name: string; value?: string }[];
+    const pageViewActionId = findLastIndex(actions, action => action.name === 'PageView');
+    if (pageViewActionId === -1) {
+      return null;
+    }
+    return actions[pageViewActionId].value;
+  }
+
+  public get location() {
+    return document.location.href;
+  }
+
+  public get referrer() {
+    return document.referrer;
+  }
+
+  private get trackingEnabled() {
+    return !!this.clientInformationProvider && !this.clientInformationProvider.disabled;
+  }
+
+  public setClientInformationProvider(clientInformationProvider: IClientInformationProvider) {
+    this.clientInformationProvider = clientInformationProvider;
```

Comment:
Could we access the ids from cookies directly instead of passing a provider to an `AnalyticsUtils` class that itself exposes getters?  Tracking could be managed elsewhere, by wiping the cookie values when analytics is inactive e.g.

```
interface IAnalyticsInformation {
    visitorId: string;
    clientId: string;
    ...
}

class AnalyticsInformation implements IAnalyticsInformation  {}
```

Reply:
Good point, my solution was getting overly complicated. I'll make a new `AnalyticsInformation` class instead.

However, since `AnalyticsInformation` won't be referencing any other class than `Cookie` from `CookieUtils`, I don't know if an interface is necessary.